bionic/code_hasher.py

```diff
@@ -93,67 +82,75 @@ def _ingest_raw_prefix_and_bytes(self, type_prefix, obj_bytes=b""):
         # The fix for this issue can also be applied to dedup code
         # references.
 
-    def _check_and_ingest(self, obj, code_context=None):
+    def _check_and_hash(self, obj, code_context=None):
         """
-        Checks for circular references before calling the _ingest
+        Checks for circular references before calling the _update_hash
         method, which does the actual encoding.
         """
         obj_id = id(obj)
 
+        # If the hash is already cached, return the cached value.
+        if obj_id in self._hashes_by_id:
+            hash_obj = self._hashes_by_id[obj_id]
```

Comment:
I'm having trouble finding where we add entries to `self._hashes_by_id`.

Reply:
This is why we have code reviews!